packages/flame/lib/src/components/base_component.dart

```diff
@@ -154,6 +154,8 @@ abstract class BaseComponent extends Component {
     if (this is HasGameRef) {
       final c = this as HasGameRef;
       gameRef ??= c.hasGameRef ? c.gameRef : null;
+    } else if (gameRef == null && isMounted) {
+      throw 'Parent was already added to Game and has no HasGameRef; in this case, gameRef is mandatory.';
     }
     if (gameRef is BaseGame) {
```

Comment:
turns out HasGameRef refers to a Game, not BaseGame. I don't think that is right, could be simplified to be only a BaseGame. now the funny thing is that this (potentially unnecessary, if we change HasGameRef) type check actually hides the null check that causes the issue I had!